pymc_marketing/mmm/base.py

```diff
@@ -280,13 +280,38 @@ def plot_components_contributions(self, **plt_kwargs: Any) -> plt.Figure:
                 az.hdi(control_contributions, hdi_prob=0.94).control_contributions
             )
 
+        if getattr(self, "yearly_seasonality", None):
+            fourier_contribution = az.extract(
+                self.fit_result,
+                var_names=["fourier_contribution"],
+                combined=False,
+            )
+            contracted_dims = [
+                d
+                for d in fourier_contribution.dims
+                if d not in ["chain", "draw", "date"]
+            ]
+            fourier_contribution = (
+                fourier_contribution.sum(contracted_dims)
+                if contracted_dims
+                else fourier_contribution
+            )
+            means.append(fourier_contribution.mean(["chain", "draw"]))
+            contribution_vars.append(
+                az.hdi(fourier_contribution, hdi_prob=0.94).fourier_contribution
+            )
```

Comment:
Maybe it makes sense to make a helper function out of this, since the exact same code appears for the `control_contributions` and even `channel_contributions`.

Reply:
You are right! Changed it in https://github.com/pymc-labs/pymc-marketing/pull/242/commits/dc5fe9418cd89425083d02f963fed63f1dcd1c92